src/equisolve/utils/split_data.py

```diff
@@ -0,0 +1,431 @@
+            group_sizes=[0.8, 0.2],  # relative, a 80% 20% train-test split
+            random_seed=100,
+        )
+
+    Split 2 tensors corresponding to input and output data into train, test, and
+    validation data. If input and output tensors have the same 10 structure
+    indices, we can split such that the train, test, and val tensors have 7,
+    2, and 1 structures in each, respectively. We want to specify absolute
+    group sizes, so will pass a list of int. Specifying the `random_seed` will
+    shuffle the structure indices before they are grouped.
+
+    .. code-block:: python
+
+        import equistore
+        from equisolve.utils import split_data
+
+        # Find the unique structure indices in the input tensor
+        unique_structure_indices = equistore.unique_metadata(
+            tensor=input, axis="samples", names=["structure"],
+        )
+        # They run from 0 -> 10 (inclusive)
+        unique_structure_indices
+        >>> Labels(
+            [(0,), (1,), (2,), (3,), (4,), (5,), (6,), (8,), (9,)],
+            dtype=[('structure', '<i4')]
+        )
+        # Verify that the output has the same unique structure indices
+        assert unique_structure_indices == equistore.unique_metadata(
+            tensor=output, axis="samples", names=["structure"],
+        )
+        >>> True
+
+        # Split the data by structure index, with an abolute split of 7, 2, 1
+        # for the train, test, and validation tensors, respectively
+        (
+            [
+                [in_train, in_test, in_val],
+                [out_train, out_test, out_val]
+            ]
+        ), grouped_labels = split_data(
+            tensors=[input, output],
+            axis="samples",
+            names=["structure"],
+            n_groups=3,  # for train-test-validation
+            group_sizes=[7, 2, 1],  # absolute; 7, 2, 1 for train, test, val
+            random_seed=100,
+        )
+        # Inspect the grouped structure indices
+        grouped_labels
+        >>> [
+            Labels(
+                [(3,), (7,), (1,), (8,), (0,), (9,), (2,)],
+                dtype=[('structure', '<i4')]
+            ),
+            Labels([(4,), (6,)], dtype=[('structure', '<i4')]),
+            Labels([(5,)], dtype=[('structure', '<i4')]),
+        ]
+    """
+    # Check input args and parse `tensors` and `names` into lists
+    tensors = [tensors] if isinstance(tensors, TensorMap) else tensors
+    names = [names] if isinstance(names, str) else names
+    _check_args(tensors, axis, names, n_groups, group_sizes, random_seed)
+
+    # Get array of unique indices to split by for each tensor in `tensors`
+    unique_idxs_list = [
+        equistore.unique_metadata(tensor, axis, names) for tensor in tensors
+    ]
+
+    # Check that the unique indices are equivalent for all input tensors
+    _check_labels_equivalent(unique_idxs_list)
+    unique_idxs = unique_idxs_list[0]
```

Comment:
Why are you picking the first one here?

Reply:
Because line 200 checks that all the Labels objects in the list are equivalent (as they should be), so we only need the first element in this list to use to split by later